src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/AttributesMustMatch.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the attributes between public members do not change.
+    /// </summary>
+    public class AttributesMustMatch : IRule
+    {
+        private readonly RuleSettings _settings;
+        private readonly HashSet<string> _attributesToExclude = new();
+
+        public AttributesMustMatch(RuleSettings settings, IRuleRegistrationContext context, IEnumerable<string>? excludeAttributesFiles)
+        {
+            _settings = settings;
+            if (excludeAttributesFiles != null)
+            {
+                readExclusions(excludeAttributesFiles);
+            }
+            context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+            context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
```

Comment:
I think we only want to run the rule right now if the exclusion file is provided. Therefore the registration should also be moved up into the if.

```suggestion
                context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
                context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
            }
```

Reply:
Why not run if there are no exclusions?
Also, doing this would require passing in an empty file to the ruleFactory in the tests.